ocsp.go

```diff
@@ -512,18 +512,18 @@ func isValidOCSPStatus(status ocspStatusCode) bool {
 // verifyPeerCertificate verifies all of certificate revocation status
 func verifyPeerCertificate(verifiedChains [][]*x509.Certificate) (err error) {
 	for i := 0; i < len(verifiedChains); i++ {
-		n := len(verifiedChains[i]) - 1
-		if !verifiedChains[i][n].IsCA || string(verifiedChains[i][n].RawIssuer) != string(verifiedChains[i][n].RawSubject) {
+		numberOfNoneRootCA := len(verifiedChains[i]) - 1
```

Comment:
Did you mean "numberofNonRootCA" ? If yes I think that would be a misnomer. Not all the certs in the chain are CA certs. There is also Leaf cert which is a cert for the Endpoint. So I'd rename it to "numberOfNonRootCerts"